tensorflow/core/protobuf/config.proto

```diff
@@ -443,6 +444,12 @@ message ConfigProto {
     // If true, use NCCL for CollectiveOps.  This feature is highly
     // experimental.
     bool collective_nccl = 7;
+    
+    // If true, any resources such as Variables used in the session will be
+    // shared with other sessions even if ClusterSpec Propagation is enabled.
+    // TODO(mrry): This field is a patch for isolate_session_state and an
+    // uniform API with flexibility in distributed runtime is needed.
+    bool share_session_state_in_clusterspec_propagation = 8;
```

Comment:
In the internal code review, @josh11b commented:

> Please document how this interacts with the isolate_session_state variable more clearly.

Please work with Josh to come up with appropriate wording here!

Reply:
@josh11b @mrry Sorry for late. How about these words as the comment for `
share_session_state_in_clusterspec_propagation`.
```text
// In default, everything works as before when cluster propagation is enabled.
// However, when `share_session_state_in_clusterspec_propagation` is turned on,
// the `isolate_session_state` field will be forced to be true even if you set
// it to false in code. Then, all resources will be shared among worker sessions
// in the same TF Server, just like the case where clusterspec propagation is
// not used in distributed training.
// TODO(mrry): This field is a patch for isolate_session_state and an
// uniform API with flexibility in distributed runtime is needed.
bool share_session_state_in_clusterspec_propagation = 8;
```

If these words satisfy your concerns, I will add them to this PR as a new commit.

Reply:
I feel like this is too complicated for unstructured prose. How about a bulleted list explaining which combinations of isolate_session_state, share_session_state_in_clusterspec_propagation, and clusterspec propagation lead to isolated vs. shared session state? Like:

* If share_session_state_in_clusterspec_propagation is true and clusterspec propagation is used then session state is shared independent of the value of isolate_session_state
* If ..., then ...
...

Right now you say things like: "when `share_session_state_in_clusterspec_propagation` is turned on, the `isolate_session_state` field will be forced to be true" which just seems backwards and confusing.

Reply:
@josh11b Get the point!
Please review the updated comment. I have pushed the change as a new commit.
Thanks a lot!

Reply:
Okay, I'm fine with this description assuming the other reviewers agree that this description actually matches what the code does.